File: ResourceManager/ViewModels/Users/UserPageViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Security
Imports ResourceManager.Configuration
Imports ResourceManager.Helpers
Imports System.Collections.ObjectModel
Imports System.Linq
Imports Microsoft.Extensions.DependencyInjection

Namespace ResourceManager.ViewModels
    Public Class UserPageViewModel
        Inherits PageViewModel
        Public Sub New()
            PageTitle = "User"
            PageIcon = AssetManager.Instance.GetImage("Users.png")
            PageColor = UserPageColor
            MyBase.HasPermission = AuthManager.User.IsInPermission(VIEW_USER_PERMISSION)

            MenuItems = New ObservableCollection(Of PageContentViewModel) From {
                Application.ServiceProvider.GetRequiredService(Of UserContentViewModel)(),
                Application.ServiceProvider.GetRequiredService(Of RoleContentViewModel)(),
                Application.ServiceProvider.GetRequiredService(Of PermissionContentViewModel)()
            }

            SelectedItem = MenuItems.FirstOrDefault()
        End Sub

    End Class
End Namespace

File: ResourceManager/ViewModels/Users/UserPageViewModel.cs
using Microsoft.Extensions.DependencyInjection;
using NetCore.Mvvm.Abstractions;
using NetCore.Security;
using ResourceManager.Configuration;
using ResourceManager.Helpers;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace ResourceManager.ViewModels
{
    public class UserPageViewModel : PageViewModel
    {
        public UserPageViewModel()
        {
            PageTitle = "User";
            PageIcon = AssetManager.Instance.GetImage("Users.png");
            PageColor = Constants.UserPageColor;
            HasPermission = AuthManager.User.IsInPermission(Constants.VIEW_USER_PERMISSION);

            MenuItems = new ObservableCollection<PageContentViewModel>
            {
                App.ServiceProvider.GetRequiredService<UserContentViewModel>(),
                App.ServiceProvider.GetRequiredService<RoleContentViewModel>(),
                App.ServiceProvider.GetRequiredService<PermissionContentViewModel>()
            };

            SelectedItem = MenuItems.FirstOrDefault();
        }
        
    }
}
